Move ContextUtils to io.opencensus.trace.unsafe package.

sebright opened the pull request:
/cc @songy23 @dinooliva

bogdandrutu commented on api/src/main/java/io/opencensus/trace/unsafe/ContextUtils.java:
```diff
@@ -11,10 +11,12 @@
 import io.grpc.Context;
 import io.opencensus.common.NonThrowingCloseable;
+import io.opencensus.trace.Span;
+import io.opencensus.trace.Tracer;
```
This is here just because of the comments. Maybe use io.opencensus.Tracer in the comments.

sebright replied:
Done

bogdandrutu commented:
You have to make methods public. It is fine because this is in an unsafe package.

sebright commented:
Sorry, I forgot to build the branch.  I split ContextUtils into two classes to avoid exposing the internal methods.  Is that okay, or would you rather also expose the other utility methods?  I added this commit:

#### Put CONTEXT_SPAN_KEY in a separate class, to avoid exposing methods in ContextUtils.

After this commit, ContextUtils is split into two classes,
io.opencensus.trace.unsafe.ContextUtils and
io.opencensus.trace.CurrentSpanUtils.  ContextUtils has one public member,
CONTEXT_SPAN_KEY.  CurrentSpanUtils has internal methods for working with the
current Span, and it is package-private.

bogdandrutu commented on api/src/main/java/io/opencensus/trace/unsafe/ContextUtils.java:
```diff
@@ -0,0 +1,31 @@
+  private ContextUtils() {}
+
+  /** The {@link io.grpc.Context.Key} used to interact with {@link io.grpc.Context}. */
+  public static final Context.Key<Span> CONTEXT_SPAN_KEY = Context.key("instrumentation-trace-key");
```
Maybe rename the key to:
opencensus-trace-key

bogdandrutu replied:
maybe: opencensus-trace-span-key?

sebright replied:
I renamed it to the second one.
